### ISSUE TITLE
Add tests for api functions

### ISSUE THREAD

**Issue by vsoch** [Kaggle/kaggle-api#58]
The API functions (kaggle.api.kaggle_api_extended) need tests to cover the majority of functions, I can give a first shot at this (locally).

**Comment by rysteboe** [comment]
Yup, this is top of my Q3 OKR's.  (I wanted it top of the Q2 OKR's but I had just started sooooo).  As always, PR's are appreciated, but I want to switch to working on tests ASAP.

**Comment by vsoch** [comment]
okey doke! Let me know if you want any help, I struggled this weekend figuring out the circle CI version 2 spec, and it (finally) started to make sense (and have some green) after 69 attempts, lol. If you don't need help, I can add the testing to the current PR when you finish up! I am really excited for working on this, I have some cool ideas for integrating the api into an "auto submission" repo via. CI. Anyway, please ping / keep me updated!

**Comment by vsoch** [comment]
Oh (what does Q3 OKRs mean?)

**Comment by rysteboe** [comment]
Ah sorry, Google (sorta?) term.  Stands for Objectives and Key Results, but basically just means objectives for some time period.  (ex. Q3 OKR's are goals for Q3, 2018 OKR's are goals for 2019 as a whole, etc).

I don't plan to hook this up to a CI build initially, but a proper CI build probably won't follow too far behind.  Unsure if we'd use Jenkins or a different solution atm, will have to chat with the team when it comes time to do that.
